lib/presto-client/index.js

```diff
@@ -95,7 +95,7 @@ Client.prototype.request = function(opts, callback) {
 
     opts = Object.assign({}, opts, client.ssl);
 
-    if (client.user)
+    if (client.user && !opts.headers[client.headers.USER])
         opts.headers[client.headers.USER] = client.user;
```

Comment:
With the addition in `statementResource`, `opts.headers` always contains `client.headers.USER` if the client is configured with the `user` option.
So, we don't set `client.user` into `opts.headers` here (just like `schema`), and it should be misleading code if exists.

Could you remove this if-clause to reduce the confusion?

Reply:
There are a few other places that use the `request` method, like the `nodes` or `query` method. Wouldn't we break the behaviour in those places as it would no longer fill X-PRESTO-USER for those?

Reply:
Oops, that's a good point! I missed other calls. My last comment was inappropriate.
So, `request` should use `client.user` in default, but `user` can be overridden per call. We can see the similar case `host` and `port` [here](https://github.com/tagomoris/presto-client-node/blob/master/lib/presto-client/index.js#L74-L78).

Could you update the patch to use `opts.user` if it is specified (and specify `req.user` [here](https://github.com/tagomoris/presto-client-node/blob/master/lib/presto-client/index.js#L227-L228)), or `client.user` in default?

Reply:
No problem 👍 

I pushed a change. Please have a look again.